File: pueblo_golf_tournament_mobile/lib/api/responses/authentication/authentication-login-dto.dart
import 'package:pueblo_golf_tournament_mobile/api/dtos/player-dto.dart';
import 'package:pueblo_golf_tournament_mobile/api/responses/base-response.dart';
import 'package:pueblo_golf_tournament_mobile/models/person.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:pueblo_golf_tournament_mobile/models/player.dart';

part 'authentication-login-dto.g.dart';

@JsonSerializable()
class AuthenticationLoginDto extends BaseResponse {
  final bool authenticated;
  final AuthenticatedUserData? data;
  AuthenticationLoginDto({
    required super.message,
    required this.authenticated,
    this.data,
  });

  factory AuthenticationLoginDto.fromJson(Map<String, dynamic> json) =>
      _$AuthenticationLoginDtoFromJson(json);

  Map<String, dynamic> toJson() => _$AuthenticationLoginDtoToJson(this);
}

@JsonSerializable()
class AuthenticatedUserData {
  final Person? personalDetails;
  final PlayerDto? playerDetails;
  final String? userName;

  AuthenticatedUserData(
      {required this.personalDetails,
      this.playerDetails,
      required this.userName});
  factory AuthenticatedUserData.fromJson(Map<String, dynamic> json) =>
      _$AuthenticatedUserDataFromJson(json);

  Map<String, dynamic> toJson() => _$AuthenticatedUserDataToJson(this);
}

File: pueblo-golf-tournament-api/Dto/Outgoing/AuthenticationLoginDto.cs
using pueblo_golf_tournament_api.Base;
using pueblo_golf_tournament_api.Entities;

namespace pueblo_golf_tournament_api.Dto.Outgoing
{
    public class AuthenticationLoginDto : BaseResponse
    {
        public AuthenticationLoginDto()
        {
            Data = new AuthenticatedUserData();
        }
        public bool Authenticated { get; set; } = false;
        public AuthenticatedUserData Data { get; set; }
    }
    public class AuthenticatedUserData
    {
        public PersonDto PersonalDetails { get; set; }
        public PlayerDto PlayerDetails { get; set; }
        public string Username { get; set; } = "";
    }
}